src/main.rs

```diff
@@ -495,3 +535,52 @@ fn main_recv(addr: &str, n_conn: &str) -> Result<()> {
 
     Ok(())
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::thread;
+
+    #[test]
+    fn test_accepts_valid_protocol() {
+        let (events_tx, events_rx) = std::sync::mpsc::channel::<SenderEvents>();
+        thread::spawn(|| {
+            std::fs::File::create("a-file").unwrap();
+            main_send("127.0.0.1:0", &["a-file".into()], 1, events_tx).unwrap();
+        });
+        match events_rx.recv().unwrap() {
+            SenderEvents::Listening(port) => {
+                main_recv(
+                    format!("127.0.0.1:{port}").as_str(),
```

Comment:
[0/5](https://handbook.mattermost.com/company/about-mattermost/list-of-terms#id-0-5-1-5-2-5-3-5-4-5-5-5) It feels a bit silly to format the string and then immediately parse it again, maybe we can make the main functions take a [`SocketAddr`](https://doc.rust-lang.org/std/net/enum.SocketAddr.html) and then in the top-level `main` we can use `SocketAddr::from_str(&args[1])`.